phpBB/adm/style/permissions.js

```diff
@@ -302,3 +317,36 @@ function set_role_settings(role_id, target_id) {
 		mark_one_option(target_id, r, (settings[r] === 1) ? 'y' : 'n');
 	}
 }
+
+function match_role_settings(id)
+{
+	var fs = document.getElementById(id),
+		cbs = fs.getElementsByTagName('input'),
+		set = {};
+
+	for (var i = 0; i < cbs.length; i++) {
+		var matches = cbs[i].id.match(/setting\[\d+]\[\d+]\[([a-z_]+)]/);
+
+		if (matches !== null && cbs[i].checked && cbs[i].value !== '-1') {
+			set[matches[1]] = parseInt(cbs[i].value);
```

Comment:
Should add the radix parameter:
```suggestion
			set[matches[1]] = parseInt(cbs[i].value, 10);
```